File: kotlin-back/src/main/kotlin/com/example/kotlinback/common/validation/ValidationUtil.java
package com.example.kotlinback.common.validation;

import com.example.kotlinback.common.exception.FormValidationException;
import org.springframework.stereotype.Component;
import org.springframework.validation.BindingResult;

import java.util.List;

@Component
public class ValidationUtil {
    public static void confirmError(BindingResult bindingResult) {
        if (bindingResult.hasErrors()) {
            StringBuilder sb = new StringBuilder();
            List<String> errorMessages = bindingResult.getAllErrors().stream().map(error -> error.getDefaultMessage()).toList();
            errorMessages.forEach(message -> sb.append("%s\n".formatted(message)));
            throw new FormValidationException(sb.toString());
        }
    }
}

File: kotlin-back/src/main/kotlin/com/example/kotlinback/common/validation/ValidationUtil.kt
package com.example.kotlinback.common.validation

import com.example.kotlinback.common.exception.FormValidationException
import org.springframework.stereotype.Component
import org.springframework.validation.BindingResult
import org.springframework.validation.ObjectError
import java.util.function.Consumer

@Component
object ValidationUtil {
    fun confirmError(bindingResult: BindingResult) {
        if (bindingResult.hasErrors()) {
            val sb = StringBuilder()
            val errorMessages = bindingResult.allErrors.stream().map { error: ObjectError -> error.defaultMessage }
                .toList()
            errorMessages.forEach(Consumer { message: String? -> sb.append("%s\n".formatted(message)) })
            throw FormValidationException(sb.toString())
        }
    }
}
